dashboard/client/src/pages/serve/ServeSystemActorDetailPage.tsx

```diff
@@ -38,27 +41,93 @@ export const ServeHttpProxyDetailPage = () => {
     );
   }
 
+  return (
+    <div>
+      <MainNavPageInfo
+        pageInfo={
+          httpProxy.node_id
+            ? {
+                id: "serveHttpProxy",
+                title: `HTTPProxyActor:${httpProxy.node_id}`,
+                pageTitle: `${httpProxy.node_id} | Serve HTTPProxyActor`,
+                path: `/serve/httpProxies/${encodeURIComponent(
+                  httpProxy.node_id,
+                )}`,
+              }
+            : {
+                id: "serveHttpProxy",
+                title: "HTTPProxyActor",
+                path: undefined,
+              }
+        }
+      />
+      <ServeSystemActorDetail
+        actor={{ type: "httpProxy", detail: httpProxy }}
+      />
+    </div>
+  );
+};
+
+export const ServeControllerDetailPage = () => {
+  const { controller, loading } = useServeControllerDetails();
+
+  if (loading) {
+    return <Loading loading />;
+  }
+
+  if (!controller) {
+    return <Typography color="error">Serve controller not found.</Typography>;
+  }
+
   return (
     <div>
       <MainNavPageInfo
         pageInfo={{
-          id: "serveHttpProxy",
-          title: `HTTPProxyActor:${httpProxy.node_id}`,
-          pageTitle: `${httpProxy.node_id} | Serve HTTPProxyActor`,
-          path: `/serve/httpProxies/${encodeURIComponent(httpProxy.node_id)}`,
+          id: "serveController",
+          title: "Serve Controller",
+          path: "/serve/controller",
         }}
       />
       <ServeSystemActorDetail
-        actor={{ type: "httpProxy", detail: httpProxy }}
+        actor={{ type: "controller", detail: controller }}
       />
     </div>
   );
 };
 
+type ActorInfo =
+  | {
+      type: "httpProxy";
+      detail: ServeHttpProxy;
+    }
+  | {
+      type: "controller";
+      detail: ServeSystemActor;
+    };
+
+type ServeSystemActorDetailProps = {
+  actor: ActorInfo;
+};
+
+export const convertActorStateForServeController = (
```

Comment:
nit: use Record/obj instead of func

Reply:
any reason for this? With the function approach, I can easily add a "catch-all" by doing an `else` block at the end.